fish-rust/src/wutil/wcstoi.rs

```diff
@@ -164,32 +230,38 @@ where
 ///  - Leading + is supported.
 pub fn fish_wcstoi<Int, Chars>(src: Chars) -> Result<Int, Error>
 where
-    Chars: Iterator<Item = char>,
+    Chars: IntoCharIter,
     Int: PrimInt,
 {
-    fish_wcstoi_impl(src, None, false)
+    fish_wcstoi_impl(src.chars(), Default::default(), &mut 0)
 }
 
 /// Convert the given wide string to an integer using the given radix.
 /// Leading whitespace is skipped.
-pub fn fish_wcstoi_radix<Int, Chars>(src: Chars, radix: u32) -> Result<Int, Error>
+pub fn fish_wcstoi_opts<Int, Chars>(src: Chars, options: Options) -> Result<Int, Error>
 where
-    Chars: Iterator<Item = char>,
+    Chars: IntoCharIter,
     Int: PrimInt,
 {
-    fish_wcstoi_impl(src, Some(radix), false)
+    fish_wcstoi_impl(src.chars(), options, &mut 0)
 }
 
-pub fn fish_wcstoi_radix_all<Int, Chars>(
+/// Convert the given wide string to an integer.
+/// The semantics here match wcstol():
+///  - Leading whitespace is skipped.
+///  - 0 means octal, 0x means hex
+///  - Leading + is supported.
+/// The number of consumed characters is returned in out_consumed.
+pub fn fish_wcstoi_partial<Int, Chars>(
     src: Chars,
-    radix: Option<u32>,
-    consume_all: bool,
+    options: Options,
+    out_consumed: &mut usize,
 ) -> Result<Int, Error>
```

Comment:
Same here re: return value vs. `&mut` parameter, although there's a risk of confusing the `Int` and `usize`.

Reply:
I prefer to match the existing C++ at this time. Refactoring can come later.